stream.js

```diff
@@ -74,19 +75,31 @@ Stream.prototype._ready = function _ready() {
   this.resume()
 }
 
-Stream.prototype._writeToSink = function _writeToSink(data) {
-  if (this.values) {
-    if (this.offsets) this.sink.write({ offset: this.cursor, value: data })
-    else this.sink.write(data)
-  } else this.sink.write(this.cursor)
+Stream.prototype._writeToSink = function _writeToSink(value, size) {
```

Comment:
Hmm, I was looking at this change and thinking: "is this slower or not?". Using the value you can get the size using `Record.size`, so instead of changing this function could you just use that function inside the new stats function?

Reply:
I know this is a hot path but I don't think it's going to get slower (we gotta measure though), because the recSize is already calculated inside log.getNextOffsetData, we are just passing around the number from there to here.